experimental/framework/tree-react-api/src/reactSharedTreeView.tsx

```diff
@@ -33,89 +32,67 @@ const SharedTree = configuredSharedTree({
-): SharedObjectKind<IReactTreeDataObject<TSchema> & IFluidLoadable & DataObject> & {
+): SharedObjectKind<
+	IReactTreeDataObject<TSchema> & IFluidLoadable & TreeDataObject<TSchema>
+> & {
 	readonly factory: IFluidDataStoreFactory;
 } {
-	class SchemaAwareTreeDataObject extends TreeDataObject<TSchema> {
-		public readonly key = key;
-		public readonly config = treeConfiguration;
-		protected readonly createInitialTree = createInitialTree;
+	class SchemaAwareTreeDataObject extends ReactTreeDataObject<TSchema> {
+		public override readonly config = treeConfiguration;
 
-		public static readonly factory = new DataObjectFactory(
-			`TreeDataObject:${key}`,
+		public static readonly factory = new PureDataObjectFactory<ReactTreeDataObject<TSchema>>(
+			`TreeDataObject`,
 			SchemaAwareTreeDataObject,
 			[SharedTree.getFactory()],
 			{},
 		);
+
+		// Populate tree with initial data on document create
+		protected override async initializingFirstTime(): Promise<void> {
+			this.tree.initialize(createInitialTree());
+		}
 	}
 	return createDataObjectKind(SchemaAwareTreeDataObject);
 }
 
 /**
- * A schema-aware Tree DataObject.
- * @remarks
- * Allows for the Tree's schema to be baked into the container schema.
+ * A schema-aware tree-backed DataObject, extended with a React Component to view the tree.
+ * @remarks Allows for the Tree's schema to be baked into the container schema.
  * @public
  */
-export interface ITreeDataObject<TSchema extends ImplicitFieldSchema> {
-	/**
-	 * The key under the root DataObject in which the {@link @fluidframework/tree#SharedTree} is stored.
-	 */
-	readonly key: string;
-
+export interface IReactTreeDataObject<TSchema extends ImplicitFieldSchema> {
 	/**
-	 * TreeViewConfiguration used to initialize new documents, as well as to interpret (schematize) existing ones.
-	 *
-	 * @remarks
-	 * The fact that a single view schema is provided here (on the data object) makes it impossible to try and apply multiple different schema.
-	 * Since the view schema currently does not provide any adapters for handling differences between view and stored schema,
-	 * its also impossible for this single view schema to handle multiple different stored schema.
-	 * Therefor, with this current API, two different applications (or different versions of the same application)
-	 * with differing stored schema requirements (as implied by their view schema) can not collaborate on the same tree.
-	 * The only schema evolution thats currently possible is upgrading the schema to one that supports a superset of what the old schema allowed,
-	 * and collaborating between clients which have view schema that exactly correspond to that stored schema.
-	 * Future work on tree as well as these utilities should address this limitation.
+	 * {@inheritDoc @fluidframework/aqueduct#TreeDataObject.config}
 	 */
 	readonly config: TreeViewConfiguration<TSchema>;
 
 	/**
-	 * The TreeView.
+	 * {@inheritDoc @fluidframework/aqueduct#TreeDataObject.tree}
```

Comment:
does cross package inheritDoc of a internal API actually work?

Reply:
It does, yes. It's currently a bit hard to detect when such references become broken though (same with `@link` comments across package boundaries). Currently such broken references are only detected when we go to generate the docs (logged as console errors). I had started writing a linter tool a while back that we could run as a post-build step to validate these and a couple other suite-level invariants API-Extractor can't cover for us, but that work was shelved.

Reply:
I think you broke the link.

Reply:
I sure did. I'll just copy the comment.

Reply:
```suggestion
	 * The schema-aware view of the tree.
```

Reply:
Every time I use `inheritDoc`, I am reminded how awful it is.